src/main/java/com/alem/todolist/controller/ToDoListController.java

```diff
@@ -1,44 +1,70 @@
 package com.alem.todolist.controller;
 
 
+import com.alem.todolist.exceptions.InvalidGroupException;
 import com.alem.todolist.repository.ToDoListRepository;
 import com.alem.todolist.model.Task;
+import com.alem.todolist.repository.UserRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.http.MediaType;
 import org.springframework.web.bind.annotation.*;
 
+import java.awt.*;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
-@EnableJpaRepositories("com.alem.todolist.repository")
+
 @RestController
 @RequestMapping("/tasks")
 public class ToDoListController {
-
     private ToDoListRepository toDoListRepository;
 
     @Autowired
-    public ToDoListController(ToDoListRepository toDoListRepository){
+    public ToDoListController(ToDoListRepository toDoListRepository) {
         this.toDoListRepository = toDoListRepository;
     }
 
-    @RequestMapping(method = RequestMethod.GET)
-    public List<Task> getTasks(){
+    @GetMapping
+    public List<Task> getTasks() {
         return this.toDoListRepository.findAll();
     }
 
-    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
-    public String printTaskDetails(@PathVariable Long id){
-        return this.toDoListRepository.findById(id).get().toString();
+
+    @GetMapping(value = "/{id}")
+    public Task printTaskDetails(@PathVariable Long id) {
+        return this.toDoListRepository.findById(id).get();
     }
 
     // task: id, desc, date, location, group
-    @PostMapping
-    public void addTask(Task task){
-        toDoListRepository.save(task);
+    @PostMapping(produces = "application/json")
+    public Task addTask(Task task) throws InvalidGroupException {
+        return toDoListRepository.save(task);
+    }
+
+    @GetMapping(value = "/forDay/{date}")
+    public List<Task> printTasksForGivenDate(@PathVariable String date){
+        String[] temp = date.split("_");
```

Comment:
I have no idea what is going on here.
1. Don't use generic variable names such as x or temp.
2. Use logger lib instead of std output.
3. Logic goes into a service layer.
4. Don't parse this by yourself. Use the already existing helpers for ISO-8601 standard.